pkg/console/operator/sync_v400.go

```diff
@@ -392,6 +400,74 @@ func (co *consoleOperator) SyncRoute(operatorConfig *operatorv1.Console) (*route
 	return rt, rtIsNew, rtErr
 }
 
+func (c *consoleOperator) SyncCustomLogoConfigMap(operatorConfig *operatorsv1.Console) (okToMount bool, err error) {
+	// validate first, to avoid a broken volume mount & a crashlooping console
+	okToMount, err = c.ValidateCustomLogo(operatorConfig)
+
+	if okToMount {
+		if err := c.UpdateCustomLogoSyncSource(operatorConfig); err != nil {
+			klog.V(4).Infoln("custom logo sync source update error")
+			return false, customerrors.NewCustomLogoError("custom logo sync source update error")
+		}
+	}
+	return okToMount, err
+}
+
+// on each pass of the operator sync loop, we need to check the
+// operator config for a custom logo.  If this has been set, then
+// we notify the resourceSyncer that it needs to start watching this
+// configmap in its own sync loop.  Note that the resourceSyncer's actual
+// sync loop will run later.  Our operator is waiting to receive
+// the copied configmap into the console namespace for a future
+// sync loop to mount into the console deployment.
+func (c *consoleOperator) UpdateCustomLogoSyncSource(operatorConfig *operatorsv1.Console) error {
+	source := resourcesynccontroller.ResourceLocation{}
+	logoConfigMapName := operatorConfig.Spec.Customization.CustomLogoFile.Name
+
+	if logoConfigMapName != "" {
```

Comment:
can this ever happen ? since `ValidateCustomLogo()` will return `false` if file name is not set.

Reply:
Fixing now.  thx!  It previously did, but I've reworked this validation a good bit to make sure its easy to read.  Aaaaand eliminated the path to allow it to cleanup that `configmap` when its not needed :)